src/HealthChecks.Elasticsearch/ElasticsearchHealthCheck.cs

```diff
@@ -32,49 +32,65 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
 
                 if (_options.AuthenticateWithBasicCredentials)
                 {
-                    settings = settings.BasicAuthentication(_options.UserName, _options.Password);
+                    if (_options.UserName is null)
+                    {
+                        throw new ArgumentNullException(nameof(_options.UserName));
+                    }
+                    if (_options.Password is null)
+                    {
+                        throw new ArgumentNullException(nameof(_options.Password));
+                    }
+                    settings = settings.Authentication(new BasicAuthentication(_options.UserName, _options.Password));
                 }
                 else if (_options.AuthenticateWithCertificate)
                 {
+                    if (_options.Certificate is null)
+                    {
+                        throw new ArgumentNullException(nameof(_options.Certificate));
+                    }
                     settings = settings.ClientCertificate(_options.Certificate);
                 }
                 else if (_options.AuthenticateWithApiKey)
                 {
-                    settings = settings.ApiKeyAuthentication(_options.ApiKeyAuthenticationCredentials);
+                    if (_options.ApiKey is null)
+                    {
+                        throw new ArgumentNullException(nameof(_options.ApiKey));
+                    }
+                    settings.Authentication(new ApiKey(_options.ApiKey));
                 }
 
                 if (_options.CertificateValidationCallback != null)
                 {
                     settings = settings.ServerCertificateValidationCallback(_options.CertificateValidationCallback);
                 }
 
-                lowLevelClient = new ElasticClient(settings);
+                elasticsearchClient = new ElasticsearchClient(settings);
 
-                if (!_connections.TryAdd(_options.Uri, lowLevelClient))
+                if (!_connections.TryAdd(_options.Uri, elasticsearchClient))
                 {
-                    lowLevelClient = _connections[_options.Uri];
+                    elasticsearchClient = _connections[_options.Uri];

```

Comment:
it's not related to your specific changes, but I wonder: does `ElasticsearchClient` implement `IDisposable`? If so, it should be disposed here (we have failed to add it to the cache and decided to reuse the existing instance)

```suggestion
                    elasticsearchClient.Dispose();
                    elasticsearchClient = _connections[_options.Uri];

```

Reply:
not implement.
Is there anything else I can do about this?